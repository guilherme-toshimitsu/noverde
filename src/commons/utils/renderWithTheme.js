import React from 'react';
import { ThemeProvider } from 'styled-components';
import defaultTheme from '../theme';
import { render } from '@testing-library/react';

function renderWithTheme(children) {
  return render(<ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>);
}

export { renderWithTheme };
