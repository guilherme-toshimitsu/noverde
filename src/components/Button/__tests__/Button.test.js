import React from 'react';

import Button from '../Button';
import { renderWithTheme } from '@commons/utils/renderWithTheme';

describe('<Button />', () => {
  it('creates a Button component', () => {
    const { asFragment } = renderWithTheme(<Button />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('creates a disabled Button component', () => {
    const { asFragment } = renderWithTheme(<Button disabled />);

    expect(asFragment).toMatchSnapshot();
  });
});
