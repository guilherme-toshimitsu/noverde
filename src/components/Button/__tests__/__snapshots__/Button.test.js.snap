// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> creates a Button component 1`] = `
<DocumentFragment>
  <button
    class="sc-bdnylx gJGOEz"
    color="primary"
    type="button"
  />
</DocumentFragment>
`;

exports[`<Button /> creates a disabled Button component 1`] = `[Function]`;
